src/main/java/com/assignment/application/other/CachingInfo.java

```diff
@@ -0,0 +1,76 @@
+package com.assignment.application.other;
+
+import com.assignment.application.entity.Employee;
+import com.assignment.application.entity.Salary;
+import com.assignment.application.repo.CompanyRepo;
+import com.assignment.application.repo.EmployeeRepo;
+import com.assignment.application.repo.SalaryRepo;
+import com.assignment.application.update.EmployeeInfoUpdate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.Caching;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class CachingInfo {
+
+    @Autowired
+    private CompanyRepo companyRepo;
+
+    @Autowired
+    private EmployeeRepo employeeRepo;
+
+    @Autowired
+    private SalaryRepo salaryRepo;
+
+    @Cacheable(value="companyCompleteInfo" , key = "#companyName" , condition = "#result==null")
+    public List<Object> getCompanyCompleteInfo(String companyName,Long companyId){
+        List<Object> companyInfoList = companyRepo.getCompDataSet(companyId);
+        return companyInfoList;
+    }
```

Comment:
Need correction in caching param keys and has a mismatch in terms of input terms.